src/cloud-api-adaptor/pkg/securecomms/ppssh/ppssh.go

```diff
@@ -0,0 +1,304 @@
+package ppssh
+
+import (
+	"bytes"
+	"context"
+	"crypto/rand"
+	"crypto/rsa"
+	"errors"
+	"fmt"
+	"log"
+	"net"
+	"os"
+	"sync"
+
+	"github.com/confidential-containers/cloud-api-adaptor/src/cloud-api-adaptor/pkg/securecomms/sshproxy"
+	"github.com/confidential-containers/cloud-api-adaptor/src/cloud-api-adaptor/pkg/securecomms/sshutil"
+	"golang.org/x/crypto/ssh"
+)
+
+const (
+	PROVEN_PP_PRIVATE_KEY_PATH  = "/tmp/provenPpPrivateKey"
+	PROVEN_WN_PUBLIC_KEY_PATH   = "/tmp/provenTePublicKey"
+	UNPROVEN_WN_PUBLIC_KEY_PATH = "/tmp/unprovenTePublicKey"
+	SIGNELTON_PATH              = "/tmp/sshSingleton"
+	PP_SID                      = "pp-sid/"
+	PP_PRIVATE_KEY              = PP_SID + "privateKey"
+	WN_PUBLIC_KEY               = "sshclient/publicKey"
+)
+
+func k8sPhase(listener net.Listener, inbounds sshproxy.Inbounds, outbounds sshproxy.Outbounds, kubernetesPhaseConfig *ssh.ServerConfig) {
+	log.Printf("Kubernetes Phase: waiting for client to connect\n")
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	var peer *sshproxy.SshPeer
+	for peer == nil {
+		nConn, err := listener.Accept()
+		if err != nil {
+			log.Fatal("failed to accept incoming connection (Kubernetes Phase): ", err)
+		}
+		//ctx, cancel := context.WithCancel(context.Background())
+		//defer cancel()
+
+		log.Printf("Kubernetes client connected\n")
+		peer, err = KubernetesSShService(ctx, nConn, kubernetesPhaseConfig)
+		if err != nil {
+			log.Printf("Failed k8sPhase: %s", err)
+			peer = nil
+			continue
+		}
+		peer.AddOutbounds(outbounds)
+		err = peer.AddInbounds(inbounds)
+		if err != nil {
+			log.Printf("Failed addInbounds during k8sPhase: %s", err)
```

Comment:
Please use `Kubernetes phase`, or maybe "cluster connection phase" throughout so that we don't need to patch it for OpenShift 😄

Reply:
Why is this not a `Fatal`?

Reply:
fun part - looking closer, this error never happens... seems to be here for historical reasons 
I will clean it.